### ISSUE TITLE
[Bug]: Provider cannot delete a Route53 SOA record if it is an updated one (bug introduced in v51.0)

### ISSUE THREAD

**Issue by mjheitland** [hashicorp/terraform-provider-aws#37863]
### Terraform Core Version

1.5.7

### AWS Provider Version

5.51.0, 5.52.0

### Affected Resource(s)

aws_route53_record

### Expected Behavior

`terraform destroy ` should delete Route53 SOA record that was once updated

### Actual Behavior

`terraform destroy` fails during deletion

### Relevant Error/Panic Output Snippet

```shell
Plan: 0 to add, 0 to change, 2 to destroy.
aws_route53_record.public_hosted_zone_soa: Destroying... [id=Z0445041PSEGKXJCF4XF_comservice-99-9999999999.aws.lpsolutions.com_SOA]
╷
│ Error: deleting Route53 Record (Z0445041PSEGKXJCF4XF_comservice-99-9999999999.aws.lpsolutions.com_SOA): operation error Route 53: ChangeResourceRecordSets, https response error StatusCode: 400, RequestID: fbced44d-4b8d-4d8c-bc11-f40bf15a2c59, InvalidChangeBatch: [A HostedZone must contain exactly one SOA record.]
```

### Terraform Configuration Files

```terraform
terraform {
  required_version = "~> 1.5.0"

  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "5.51 # workaround:      version = ">= 5.0, < 5.51" # 5.51 introduced a bug: Route53 SOA records cannot be deleted any longer! Please update this constraint once the bug is resolved.
    }
  }
}

provider "aws" {
  region = "eu-central-1"
  default_tags {
    tags = {
      Project      = "tftest"
      Environment  = "saas-dev"
      "managed-by" = "terraform"
    }
  }
}

resource "aws_route53_zone" "public_hosted_zone" {
  name = "comservice-99-9999999999.aws.lpsolutions.com"
}

resource "aws_route53_record" "public_hosted_zone_soa" {
  allow_overwrite = true
  zone_id         = aws_route53_zone.public_hosted_zone.id
  name            = aws_route53_zone.public_hosted_zone.name
  type            = "SOA"
  ttl             = 900

  //noinspection HILUnresolvedReference
  records = [
    # [authority-domain] [hostmaster-email-address] [zone-serial-number] [refresh-time] [retry-time] [expire-time] [negative caching TTL]
    format("%s. awsdns-hostmaster.amazon.com. 1 7200 900 1209600 60", aws_route53_zone.public_hosted_zone.primary_name_server)
  ]
}
```

### Steps to Reproduce

1. set your AWS credentials
2. copy the attached configuration into an empty folder, navigate into the folder
3. rm -rf .terraform
4. set constraint to aws provider version that you want to test (e.g. v
5. terraform init
6. terraform apply # will succeed
7. terraform destroy # will return an error if the provider version is still buggy

### Important Factoids

Consider this line in the SOA record:
allow_override = true

### Would you like to implement a fix?

None

**Comment by ewbankkit** [comment]
I _think_ this will be addressed via https://github.com/hashicorp/terraform-provider-aws/pull/37850. Will investigate.

**Comment by ewbankkit** [comment]
Verified that this is fixed by #37850 and will be available in Terraform AWS Provider **v5.53.0**, likely released later today.

#### v5.52.0

```
Do you really want to destroy all resources?
  Terraform will destroy all your managed infrastructure, as shown above.
  There is no undo. Only 'yes' will be accepted to confirm.

  Enter a value: yes

aws_route53_record.public_hosted_zone_soa: Destroying... [id=Z09159933EE0YK3LTOO5W_comservice-99-9999999999.aws.lpsolutions.com_SOA]
╷
│ Error: deleting Route53 Record (Z09159933EE0YK3LTOO5W_comservice-99-9999999999.aws.lpsolutions.com_SOA): operation error Route 53: ChangeResourceRecordSets, https response error StatusCode: 400, RequestID: b7c3e67e-dd0f-4fc0-8560-5289668a962a, InvalidChangeBatch: [A HostedZone must contain exactly one SOA record.]
│ 
│ 
╵
```
